### PR TITLE
[BPK-1588] Add image to web nav bar

### PR DESCRIPTION
**Pull request by georgegillams**

### CODE REVIEW

File: packages/bpk-docs/src/components/neo/DocsPageWrapper/DocsPageWrapper.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ class DocsPageWrapper extends React.Component {
 
   constructor(props) {
     super(props);
-    this.state = { selected: this.props.nativeSubpage ? 'native' : 'web' };
+    this.state = { selected: 'web' };
```
Comment (georgegillams) [inline packages/bpk-docs/src/components/neo/DocsPageWrapper/DocsPageWrapper.js]: undo this!

File: packages/bpk-docs/src/pages/NavigationBarPage/NavigationBarPage.js [hunk 2]
```diff
@@ -118,6 +118,65 @@ const components = [
...
+    title: 'With icon in title',
+    blurb: [
+      <Paragraph>The title can include elements other than text.</Paragraph>,
+    ],
+    examples: [
+      <div className={getClassNames('bpk-navigation-bar-example')}>
+        <BpkNavigationBar
+          id="default-bpk-nav"
+          title={
+            <div style={{ display: 'flex', flexDirection: 'row' }}>
```
Comment (georgegillams) [inline packages/bpk-docs/src/pages/NavigationBarPage/NavigationBarPage.js]: replace with an image

File: packages/bpk-component-navigation-bar/src/BpkNavigationBar.js [hunk 3]
```diff
@@ -72,7 +72,7 @@ const BpkNavigationBar = (props: Props) => {
 };
 
 BpkNavigationBar.propTypes = {
-  title: PropTypes.string.isRequired,
+  title: PropTypes.node.isRequired,
```
Comment (tiagoengel) [inline packages/bpk-component-navigation-bar/src/BpkNavigationBar.js]: this should be `element`

File: packages/bpk-component-navigation-bar/stories.js [hunk 4]
```diff
@@ -107,4 +107,35 @@ storiesOf('bpk-component-navigation-bar', module)
...
       />
     </div>
+  ))
+  .add('With icon', () => (
+    <div className={getClassNames('bpk-navigation-bar-story')}>
+      <BpkNavigationBar
+        id="test"
+        title={
+          <div style={{ display: 'flex', alignItems: 'center' }}>
+            <img
```
Comment (tiagoengel) [inline packages/bpk-component-navigation-bar/stories.js]: Wouldn't it be easier to use `BpkIcon`. If users want to use something else they can but as an example I think we should use an icon.

File: packages/bpk-component-navigation-bar/stories.js [hunk 5]
```diff
@@ -107,4 +107,35 @@ storiesOf('bpk-component-navigation-bar', module)
         }
       />
     </div>
+  ))
+  .add('With icon', () => (
+    <div className={getClassNames('bpk-navigation-bar-story')}>
+      <BpkNavigationBar
+        id="test"
+        title={
+          <div style={{ display: 'flex', alignItems: 'center' }}>
```
Comment (tiagoengel) [inline packages/bpk-component-navigation-bar/stories.js]: can we get rid of the inline styles and use a class? Also, I think a `span` is semantically more appropriate here.

File: packages/bpk-component-navigation-bar/src/BpkNavigationBar.js [hunk 6]
```diff
@@ -29,7 +29,7 @@ const getClassNames = cssModules(STYLES);
 
 export type Props = {
   id: string,
-  title: string,
+  title: string | Element<any>,
```
Comment (tiagoengel) [inline packages/bpk-component-navigation-bar/src/BpkNavigationBar.js]: can we add a new test that renders with an element instead of a string?

Comment (georgegillams) [reply packages/bpk-component-navigation-bar/src/BpkNavigationBar.js]: Done 👍

File: packages/bpk-component-navigation-bar/readme.md [hunk 7]
```diff
@@ -44,7 +44,7 @@ export default () => (
 | Property              | PropType | Required | Default Value    |
 | --------------------- | -------- | -------- | ---------------- |
 | id                    | string   | true     | -                |
-| title                 | string   | true     | -                |
+| title                 | node     | true     | -                |
```
Comment (tiagoengel) [inline packages/bpk-component-navigation-bar/readme.md]: this needs to be changed to `element` to

File: unreleased.md [hunk 8]
```diff
@@ -3,3 +3,6 @@
 **Added:**
 - react-native-bpk-component-pagination-dots:
   - Introducing the React Native pagination dots component.
+
+- bpk-component-navigation-bar:
+  - The title prop now accepts nodes, so images and other elements can be passed in as well as text.
```
Comment (tiagoengel) [inline unreleased.md]: "accepts elements"

**Comment by tiagoengel** [comment]
@georgegillams I've talked with @jamesf3rguson and we're going to change a little bit the logo to use only one image and no text. That reflects a little bit better the use case they're expecting from it. But otherwise looks good.

**Pull request merged**